### ISSUE TITLE
Loader doesn't follow PSR-4 specification

### ISSUE THREAD

**Issue by luiz-brandao-jr** [phalcon/cphalcon#11021]
I ran into a problem and I realized that the Loader doesn't allow a namespace to be mapped to multiple directories. Take this [PSR-4 example](http://www.php-fig.org/psr/psr-4/examples/) for instance:

``` php
$loader->addNamespace('Foo\Bar', '/path/to/packages/foo-bar/src');
$loader->addNamespace('Foo\Bar', '/path/to/packages/foo-bar/tests');
```

This is currently not possible with Phalcon\Loader because it threats the namespaces array as unidimensional. Is this a know issue? Any plans to fix that?

**Comment by sergeyklay** [comment]
Yes, we plan to fix it. Something like this:
```
$loader->addNamespace('Foo\Bar', [ '/path/to/packages/foo-bar/src']);

$loader->addNamespace(
    'Buz\Bar',
    [
         '/path/to/packages/buz-bar/src1',
         '/path/to/packages/buz-bar/src2',
          '/path/to/packages/buz-bar/src3'
    ]
);
```

**Comment by sergeyklay** [comment]
@noobiwan I've introduced `Phalcon\Loader\Extended` in the [Phalcon Incubator](https://github.com/phalcon/incubator/pull/457). This component extends `Phalcon\Loader` and adds ability to set multiple directories a namespace.

Also I have [tried to implement](https://github.com/phalcon/cphalcon/issues/11044) this directly in the Phalcon. But we have some difficulties with the latest version of Zephir and Phalcon 2.1 . This will be fixed later.

However, I would like to draw your attention that even if we fix this PR or/and some Zephir issues, the ability that you described will be implemented in Phalcon 2.1, not in 2.0.x

**Comment by butamuh4o** [comment]
Thank you @sergeyklay
